elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/QueryValidator.java

```diff
@@ -0,0 +1,156 @@
+                if (metrics.stream().noneMatch(m -> m.getAlias().equals(fieldName))) {
+                    throw new InvalidOperationException(
+                            String.format(
+                                    "Metric field %s must be aggregated before filtering in having clause.",
+                                    fieldName));
+                }
+            } else {
+                if (dimensionProjections.stream().noneMatch(dim -> dim.getAlias().equals(fieldName))) {
+                    throw new InvalidOperationException(
+                            String.format(
+                                    "Dimension field %s must be grouped before filtering in having clause.",
+                                    fieldName));
+                }
+            }
+        } else if (havingClause instanceof AndFilterExpression) {
+            validateHavingClause(((AndFilterExpression) havingClause).getLeft());
+            validateHavingClause(((AndFilterExpression) havingClause).getRight());
+        } else if (havingClause instanceof OrFilterExpression) {
+            validateHavingClause(((OrFilterExpression) havingClause).getLeft());
+            validateHavingClause(((OrFilterExpression) havingClause).getRight());
+        } else if (havingClause instanceof NotFilterExpression) {
+            validateHavingClause(((NotFilterExpression) havingClause).getNegated());
+        }
+    }
+
+    /**
+     * Method to verify that all the sorting options provided
+     * by the user are valid and supported.
+     */
+    public void validateSorting() {
+        Sorting sorting = query.getSorting();
+        if (sorting == null) {
+            return;
+        }
+        Map<Path, Sorting.SortOrder> sortClauses = sorting.getValidSortingRules(type, dictionary);
+        sortClauses.keySet().forEach((path) -> validateSortingPath(path, allFields));
+    }
+
+    /**
+     * Verifies that the current path can be sorted on
+     * @param path The path that we are validating
+     * @param allFields Set of all field names included in initial query
+     */
+    private void validateSortingPath(Path path, Set<String> allFields) {
+        List<Path.PathElement> pathElemenets = path.getPathElements();
+
+        //TODO add support for double nested sorting
+        if (pathElemenets.size() > 2) {
+            throw new UnsupportedOperationException(
+                    "Currently sorting on double nested fields is not supported");
+        }
+        Path.PathElement currentElement = pathElemenets.get(0);
+        String currentField = currentElement.getFieldName();
+        Class<?> currentClass = currentElement.getType();
+        if (!allFields.stream().anyMatch(field -> field.equals(currentField))) {
+            throw new InvalidOperationException("Can't sort on " + currentField + " as it is not present in query");
+        }
+        if (dictionary.getIdFieldName(currentClass).equals(currentField)
```

Comment:
flip this expression `dictionary.getIdFieldName(currentClass).equals(currentField)` to avoid the case that `getIdFieldName()` returns `null`